eth/db/typing.py

```diff
@@ -0,0 +1 @@
+JournalDBCheckpoint = int
```

Comment:
Shouldn't  this use [`typing.NewType`](https://docs.python.org/3/library/typing.html#newtype)?  My impression is that is the *correct* way to do this.

Reply:
`JournalDBCheckpoint = int` is just an alias usually used to get a more descriptive name to a type that would otherwise be cumbersome to use repetitively. From the perspective of the type system the alias is identical to the type it is aliased to.

On the other hand [`typing.NewType`](https://docs.python.org/3/library/typing.html#newtype), really creates a different kind of type so that the type system can enforce it isn't mixed up with a different type somewhere else. In other words `JournalDBCheckpoint` would become entirely different from `int` and hence can `JournalDBCheckpoint(1) + 1` would be rejected from the type system.

I guess a [`typing.NewType`](https://docs.python.org/3/library/typing.html#newtype) was meant to be used here.